perception/sub8_batcave_ogrid/src/OGridGen.cpp

```diff
@@ -0,0 +1,142 @@
+#include "OGridGen.hpp"
+
+//TODO: Add service call to clear ogrid
+
+OGridGen::OGridGen() : nh(ros::this_node::getName()), classification(&nh)
+{
+  pubGrid = nh.advertise<nav_msgs::OccupancyGrid>("/ogrid", 10, true);
+  pubPointCloud = nh.advertise<sensor_msgs::PointCloud2>("/point_cloud", 1);
+
+  nh.param<float>("resolution", resolution, 0.2f);
+  nh.param<float>("ogrid_size", ogrid_size, 50.f);
+  nh.param<float>("pool_depth", pool_depth, 7.f);
+  
+  int pointCloudBufferSize;
+  nh.param<int>("buffer_size", pointCloudBufferSize, 5000);
+  pointCloudBuffer.resize(pointCloudBufferSize);
+
+  get_bounds = nh.serviceClient<sub8_msgs::Bounds>("get_bounds");
+
+  timer = nh.createTimer(ros::Duration(0.3), std::bind(&OGridGen::publish_ogrid, this, std::placeholders::_1));
+  sub = nh.subscribe("/blueview_driver/ranges", 1, &OGridGen::callback, this);
+}
+
+void OGridGen::publish_ogrid(const ros::TimerEvent &)
+{
+  // Draw ogrid
+  sub8_msgs::Bounds bound_hold;
+  if (get_bounds.call(bound_hold))
+  {
+    bounds = std::vector<cv::Point>(4);
+    size_t k = 0;
+    for (auto &p : bound_hold.response.bounds)
+    {
+      bounds.at(k) = (cv::Point(p.x / resolution + matOgrid.cols / 2, p.y / resolution + matOgrid.rows / 2));
+      k++;
+    }
+    const cv::Point *pts = (const cv::Point *)cv::Mat(bounds).data;
+    int npts = cv::Mat(bounds).rows;
+    cv::polylines(matOgrid, &pts, &npts, 1, true, 255, 3, CV_8U, 0);
```

Comment:
I think someting like this would be cleaner (You don't wan't your c++ to look too much like old c):
> auto* polygons = &bounds;
> auto npts = bounds.size();
> cv::polylines(matOgrid, &polygons, &npts, 1, true, 255, 3, CV_8U);